public/app/plugins/datasource/influxdb/datasource.test.ts

```diff
@@ -0,0 +1,319 @@
+import { lastValueFrom, of } from 'rxjs';
+
+import { ScopedVars } from '@grafana/data';
+import { BackendSrvRequest } from '@grafana/runtime/';
+import config from 'app/core/config';
+
+import { TemplateSrv } from '../../../features/templating/template_srv';
+
+import { BROWSER_MODE_DISABLED_MESSAGE } from './constants';
+import InfluxDatasource from './datasource';
+import {
+  getMockDSInstanceSettings,
+  getMockInfluxDS,
+  mockBackendService,
+  mockInfluxFetchResponse,
+  mockInfluxQueryRequest,
+  mockInfluxQueryWithTemplateVars,
+  mockTemplateSrv,
+} from './mocks';
+import { InfluxQuery, InfluxVersion } from './types';
+
+// we want only frontend mode in this file
+config.featureToggles.influxdbBackendMigration = false;
+const fetchMock = mockBackendService(mockInfluxFetchResponse());
+
+describe('InfluxDataSource Frontend Mode', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should throw an error if there is 200 response with error', async () => {
+    const ds = getMockInfluxDS();
+    fetchMock.mockImplementation(() => {
+      return of({
+        data: {
+          results: [
+            {
+              error: 'Query timeout',
+            },
+          ],
+        },
+      });
+    });
+
+    try {
+      await lastValueFrom(ds.query(mockInfluxQueryRequest()));
+    } catch (err) {
+      if (err instanceof Error) {
+        expect(err.message).toBe('InfluxDB Error: Query timeout');
+      }
+    }
+  });
+
+  describe('outdated browser mode', () => {
+    it('should throw an error when querying data', async () => {
+      const instanceSettings = getMockDSInstanceSettings();
+      instanceSettings.access = 'direct';
+      const ds = getMockInfluxDS(instanceSettings);
+      try {
+        await lastValueFrom(ds.query(mockInfluxQueryRequest()));
+      } catch (err) {
+        if (err instanceof Error) {
```

Comment:
if this didn't err for some reason, I don't think this test would fail, I think it would just fail to assert? 
Should we expect that the test has asserted?
https://jestjs.io/docs/expect#expectassertionsnumber

Reply:
The test above it as well, but they appear to be the only tests with conditional assertions